catalog/plugin/catalog-plugin-expirationdate/src/test/java/org/codice/ddf/catalog/plugin/expiration/TestExpirationDatePlugin.java

```diff
@@ -0,0 +1,597 @@
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.StopProcessingException;
+
+public class TestExpirationDatePlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestExpirationDatePlugin.class);
+
+    private static final String DATE_FORMAT = "MM-dd-yyyy HH:mm:ss.SSS";
+
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormat
+            .forPattern(DATE_FORMAT);
+
+    private static final DateTime MODIFIED_DATE = DATE_TIME_FORMATTER
+            .parseDateTime("09-01-1990 00:00:00.000");
+
+    private static final DateTime EXPIRATION_DATE = DATE_TIME_FORMATTER
+            .parseDateTime("09-10-1990 00:00:00.000");
+
+    @Test
+    /**
+     * Tests the process(CreateRequest createRequest) method.
+     *
+     * Overwrite the 'blank' expiration date (single metacard).  The new expiration date is 10 after the modified date.
+     *
+     * offsetFromModifiedDate: 10
+     * overwriteIfBlank: true
+     * overwriteIfExists: false
+     *
+     * @throws PluginExecutionException
+     * @throws StopProcessingException
+     */ public void testProcessCreateRequestMetacardHasNoExpirationDateOverwriteIfBlankIsTrueOverwriteIfExistsIsFalseDaysOffsetFromModifiedDateIs10()
+            throws PluginExecutionException, StopProcessingException {
+
+        // Setup
+        ExpirationDatePlugin expirationDatePlugin = new ExpirationDatePlugin();
+
+        // Items configured via the admin console
+        int days = 10;
+        expirationDatePlugin.setOffsetFromModifedDate(days);
+        expirationDatePlugin.setOverwriteIfBlank(true);
+        expirationDatePlugin.setOverwriteIfExists(false);
+
+        CreateRequest mockCreateRequest = mock(CreateRequest.class);
+        int numberOfMetacards = 1;
+        List<Metacard> mockMetacards = createMockMetacardsWithNoExpirationDate(numberOfMetacards);
+        when(mockCreateRequest.getMetacards()).thenReturn(mockMetacards);
+
+        // Perform Test
+        expirationDatePlugin.process(mockCreateRequest);
+
+        // Verify
+        assertThat(mockCreateRequest.getMetacards().size(), is(numberOfMetacards));
```

Comment:
There's also a [`hasSize()`](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#hasSize(int)) matcher you can use for matching collection sizes.